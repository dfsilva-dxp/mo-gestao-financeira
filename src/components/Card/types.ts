export type CardProps = {
  balance: string;
};
