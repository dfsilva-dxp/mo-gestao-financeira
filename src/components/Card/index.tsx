import Image from "next/image";

import Logo from "components/Logo";

import {
  Balance,
  CardFooter,
  CardInfo,
  CardNumber,
  Info,
  Title,
  Wrapper
} from "./styles";
import { CardProps } from "./types";

export default function Card({ balance }: CardProps) {
  return (
    <Wrapper>
      <CardInfo>
        <Info>
          <Title>Carteira</Title>
          <Balance>{balance}</Balance>
        </Info>

        <Logo size="small" />
      </CardInfo>

      <CardFooter>
        <CardNumber>**** **** **** 2376</CardNumber>
        <Image
          src="/img/card-brand.svg"
          aria-label="Card Brand"
          width="35"
          height="20"
        />
      </CardFooter>
    </Wrapper>
  );
}
