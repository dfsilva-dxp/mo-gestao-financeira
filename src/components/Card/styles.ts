import styled, { css } from "styled-components";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background: url("/img/looper.svg") center center/cover no-repeat,
      ${theme.colors.gradient};
    width: 34rem;
    padding: ${theme.spacings.small} ${theme.spacings.large};
    border-radius: ${theme.border.radius.default};
    display: grid;
    gap: ${theme.spacings.small};
  `}
`;

export const CardInfo = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const Info = styled.div`
  ${({ theme }) => css`
    display: grid;
    gap: ${theme.spacings.xxsmall};
  `}
`;

export const Title = styled.h4`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    color: ${theme.colors.gray};
  `}
`;

export const Balance = styled.strong`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.large};
  `}
`;

export const CardFooter = styled.footer`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const CardNumber = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xsmall};
    font-weight: ${theme.font.weight.bold};
    color: ${theme.colors.gray};
  `}
`;
