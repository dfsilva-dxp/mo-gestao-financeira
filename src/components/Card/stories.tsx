import { Story, Meta } from "@storybook/react/types-6-0";
import { ComponentProps } from "react";

import Card from ".";

export default {
  title: "Components/Card",
  component: Card,
  args: {
    balance: "R$1060,00"
  },
  parameters: {
    layout: "centered"
  }
} as Meta;

export const Basic: Story<ComponentProps<typeof Card>> = ({ ...props }) => (
  <Card {...props} />
);
