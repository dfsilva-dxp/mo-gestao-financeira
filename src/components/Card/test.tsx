import { screen } from "@testing-library/react";

import { renderWithTheme } from "utils/tests/helpers";

import Card from ".";

describe("<Card />", () => {
  it("should render currectly", () => {
    const { container } = renderWithTheme(<Card balance="R$1060,00" />);

    expect(
      screen.getByRole("heading", { name: /carteira/i })
    ).toBeInTheDocument();

    expect(screen.getByText("R$1060,00")).toBeInTheDocument();

    expect(screen.getByRole("img", { name: /mo app/i })).toBeInTheDocument();

    expect(
      screen.getByRole("img", { name: /mo app/i }).parentElement
    ).toHaveStyle({
      width: "4rem"
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
