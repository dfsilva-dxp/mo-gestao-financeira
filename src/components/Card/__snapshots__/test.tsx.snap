// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render currectly 1`] = `
.c5 {
  color: #fff;
  width: 4rem;
}

.c0 {
  background: url("/img/looper.svg") center center/cover no-repeat,linear-gradient(145deg,rgba(33,34,45,1) 0%,rgba(58,44,58,1) 100%);
  width: 34rem;
  padding: 2.0rem 4.0rem;
  border-radius: 1.6rem;
  display: grid;
  gap: 2.0rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  display: grid;
  gap: 0.8rem;
}

.c3 {
  font-size: 1.6rem;
  color: #797D9A;
}

.c4 {
  font-size: 2.2rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 {
  font-size: 1.2rem;
  font-weight: 700;
  color: #797D9A;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h4
        class="c3"
      >
        Carteira
      </h4>
      <strong
        class="c4"
      >
        R$1060,00
      </strong>
    </div>
    <div
      class="c5"
      color="white"
    >
      <svg
        aria-label="Mo app"
        fill="none"
        role="img"
        viewBox="0 0 65 42"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="17.075"
          cy="24.664"
          fill="#F9769D"
          r="17.075"
        />
        <path
          d="M45.534 5.692a5.692 5.692 0 1 1-11.384 0 5.692 5.692 0 0 1 11.384 0Z"
          fill="#797D9A"
        />
        <path
          d="M14.972 16.362c.164 0 .327.035.491.105.187.07.339.187.456.35l9.677 14.516-.947-.035 9.712-14.48c.257-.35.573-.503.947-.456a.9.9 0 0 1 .7.315c.211.187.316.433.316.737v22.439c0 .28-.105.526-.315.736-.21.21-.468.316-.772.316-.327 0-.596-.105-.806-.316a1.01 1.01 0 0 1-.316-.736v-20.3l.632.07-8.87 13.428a.904.904 0 0 1-.316.35c-.14.07-.304.094-.491.07a.903.903 0 0 1-.491-.07 1.253 1.253 0 0 1-.35-.35l-8.906-13.569.666-.456v20.827c0 .28-.105.526-.316.736a.938.938 0 0 1-.736.316.938.938 0 0 1-.736-.316 1.074 1.074 0 0 1-.28-.736v-22.44c0-.28.093-.525.28-.736.21-.21.467-.315.771-.315ZM58.944 31.929c0 1.776-.397 3.378-1.192 4.803a8.824 8.824 0 0 1-3.156 3.331c-1.332.795-2.84 1.192-4.522 1.192-1.66 0-3.168-.397-4.523-1.192-1.333-.818-2.396-1.928-3.19-3.33-.772-1.426-1.158-3.028-1.158-4.804 0-1.8.386-3.4 1.157-4.803.795-1.403 1.858-2.513 3.19-3.331 1.356-.818 2.864-1.227 4.524-1.227 1.683 0 3.19.409 4.522 1.227a8.824 8.824 0 0 1 3.156 3.33c.795 1.403 1.192 3.004 1.192 4.804Zm-2.104 0c0-1.426-.292-2.688-.876-3.787-.584-1.122-1.391-1.998-2.42-2.63-1.005-.654-2.161-.98-3.47-.98-1.286 0-2.443.326-3.471.98-1.006.632-1.812 1.508-2.42 2.63-.584 1.099-.876 2.361-.876 3.787 0 1.426.292 2.688.876 3.787.608 1.098 1.414 1.975 2.42 2.63 1.028.63 2.185.946 3.47.946 1.31 0 2.466-.316 3.472-.947a6.9 6.9 0 0 0 2.419-2.63c.584-1.098.876-2.36.876-3.786ZM63.527 40.905c-.397 0-.7-.129-.911-.386-.21-.257-.316-.584-.316-.982v-.736c0-.42.106-.748.316-.982.21-.233.514-.35.911-.35h.246c.397 0 .701.117.911.35.21.234.316.561.316.982v.736c0 .398-.105.725-.316.982-.21.257-.514.386-.911.386h-.246Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <footer
    class="c6"
  >
    <p
      class="c7"
    >
      **** **** **** 2376
    </p>
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2735%27%20height=%2720%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        aria-label="Card Brand"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </footer>
</div>
`;
